.github/workflows/scripts/npm_release_files/ts_serialize.d.ts

```diff
@@ -1,6 +1,16 @@
 declare module "@gamebridgeai/ts_serialize" {
+  /** to be implemented by external authors on their models  */
+  export interface TransformKey {
+    /** a function that will be called against
+   * every property key transforming the key
+   * with the provided function
+   */
+    tsTransformKey(key: string): string;
```

Comment:
Maybe we need two - one for ser and one to deser. (Camel to snake vs snake to camel)

Reply:
We always have the original property key so the transform only needs to be one way.